app/models/models.py

```diff
@@ -14,85 +11,209 @@
+    group = db.Column(db.Integer, db.ForeignKey('group.id'))
+    path = db.Column(db.String, nullable=False)
+    text = db.Column(db.Text, nullable=False)
+    json = db.Column(db.JSON, nullable=False)
+    created = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    owner_r = db.Column(db.Boolean, nullable=False, default=True)
+    owner_w = db.Column(db.Boolean, nullable=False, default=True)
+    owner_x = db.Column(db.Boolean, nullable=False, default=True)
+    group_r = db.Column(db.Boolean, nullable=False, default=True)
+    group_w = db.Column(db.Boolean, nullable=False, default=False)
+    group_x = db.Column(db.Boolean, nullable=False, default=True)
+    world_r = db.Column(db.Boolean, nullable=False, default=False)
+    world_w = db.Column(db.Boolean, nullable=False, default=False)
+    world_x = db.Column(db.Boolean, nullable=False, default=False)
+
+    job = db.relationship('Job', back_populates='flowchart', lazy=True)
+    projects = db.relationship(
+        'Project', secondary=flowchart_project, back_populates='flowcharts'
+    )
 
     def __repr__(self):
-        return F"Job(path={self.path}, flowchart_id={self.flowchart_id}, submission_date={self.submission_date})"
+        return F'Flowchart(id={self.id}, description={self.description}, path={self.path})'  # noqa: E501
 
-class Project(db.Model):
-    __tablename__ = "projects"
 
-    name = db.Column(db.String, nullable=False, primary_key=True)
-    project_path = db.Column(db.String, nullable=False, primary_key=True)
-    description = db.Column(db.String, nullable=True, unique=False)
+class Job(db.Model):
+    __tablename__ = 'job'
+
+    id = db.Column(db.Integer, primary_key=True)
+    flowchart_id = db.Column(db.String(32), db.ForeignKey('flowchart.id'))
+    title = db.Column(db.String, nullable=True)
+    description = db.Column(db.Text, nullable=True)
+    owner = db.Column(db.Integer, db.ForeignKey('user.id'))
+    group = db.Column(db.Integer, db.ForeignKey('group.id'))
+    path = db.Column(db.String, nullable=False, unique=True)
+    submitted = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    started = db.Column(db.DateTime)
+    finished = db.Column(db.DateTime)
+    status = db.Column(db.String, nullable=False, default='imported')
+    owner_r = db.Column(db.Boolean, nullable=False, default=True)
+    owner_w = db.Column(db.Boolean, nullable=False, default=True)
+    owner_x = db.Column(db.Boolean, nullable=False, default=True)
+    group_r = db.Column(db.Boolean, nullable=False, default=True)
+    group_w = db.Column(db.Boolean, nullable=False, default=False)
+    group_x = db.Column(db.Boolean, nullable=False, default=True)
+    world_r = db.Column(db.Boolean, nullable=False, default=False)
+    world_w = db.Column(db.Boolean, nullable=False, default=False)
+    world_x = db.Column(db.Boolean, nullable=False, default=False)
+
+    flowchart = db.relationship('Flowchart', back_populates='job')
```

Comment:
If we change `job` to `jobs` in the Flowchart table, we would change this line to `back_populates='jobs'`

Reply:
Done